src/instructlab/model/train.py

```diff
@@ -1042,6 +1049,47 @@ def _evaluate_dir_of_checkpoints(
     return TrainingJournal.best_checkpoint(phase_model=phase_model)
 
 
+def _run_phase(
+    train_args: TrainingArgs,
+    torch_args: TorchrunArgs,
+    data_path: pathlib.Path,
+    checkpoint_dir: pathlib.Path,
+    num_epochs: int | None,
+    samples_per_save: int | None,
+    effective_batch_size: int | None,
+    journal: TrainingJournal,
+    phase_model: TrainPhaseModel | EvalPhaseModel,
+    next_phase: TrainingPhases,
+    model_override: pathlib.Path | None = None,
+):
+    failed = False
+    try:
+        _training_phase(
+            train_args=train_args.model_copy(deep=True),
+            torch_args=torch_args,
+            data_path=data_path,
+            checkpoint_dir=checkpoint_dir,
+            num_epochs=num_epochs,
+            samples_per_save=samples_per_save,
+            effective_batch_size=effective_batch_size,
+            model_override=model_override,
+            # model override not necessary because we expect model to come from ctx.params.model_path.
+        )
+    except RuntimeError as e:
+        logger.error("Failed during training loop: ", e)
+        failed = True
+    else:
+        # progress onto the next phase on success
+        journal.current_phase = next_phase
+    finally:
+        phase_model.ended_at_utc = TrainingJournal.now_utc()
+        journal.commit()
+
+    if failed:
+        logger.info("Phase failed, exiting...")
```

Comment:
I think this should be a click.echo or a print.  The line being, if logging was disabled or sent to a log, would you still want this message to be printed.

Reply:
The logger is used here since I saw it being used elsewhere in this script. I'll update it to use click.secho though for broader consistency.